Source/Core/VideoCommon/Present.cpp

```diff
@@ -620,13 +661,19 @@ std::tuple<int, int> Presenter::CalculateOutputDimensions(int width, int height,
   if (!allow_stretch && aspect_mode == AspectMode::Stretch)
     aspect_mode = AspectMode::Auto;
 
-  // Find the closest integer aspect ratio,
-  // this avoids a small black line from being drawn on one of the four edges
   if (!g_ActiveConfig.bCrop && aspect_mode != AspectMode::Stretch)
   {
+    // Find the closest integer aspect ratio,
+    // this avoids a small black line from being drawn on one of the four edges
     const float draw_aspect_ratio = CalculateDrawAspectRatio(allow_stretch);
-    const auto [int_width, int_height] =
+    auto [int_width, int_height] =
         FindClosestIntegerResolution(scaled_width, scaled_height, draw_aspect_ratio);
+    // Given this is a bit of a random thing to do, we only do it if the user has the
+    // auto aspect ratio mode selected.
+    if (aspect_mode == AspectMode::Auto)
+    {
+      TryToSnapToXFBSize(int_width, int_height, m_xfb_rect.GetWidth(), m_xfb_rect.GetHeight());
+    }
```

Comment:
I disagree with this only happening on Auto, the 'Force 4:3' and 'Force 16:9' options should IMO always behave exactly like 'Auto' but without the heuristic for what aspect ratio the game intends. Otherwise you get into really annoying situations where the heuristic fails but you can't avoid the blurriness from being a pixel off.

I also think that this logic should just go into `FindClosestIntegerResolution()`, ie the resolution returned by `FindClosestIntegerResolution()` should be the output image resolution.

Reply:
`FindClosestIntegerResolution()` is just a math function. I had merged the two functions together in a prior version of the PR but then changed it to allow for case by case handling, as it seemed more appropriate and flexible.

Reply:
You can always avoid the blurriness by resizing the window yourself. This is only ever really a problem when using the setting to automatically resize the game window based on the internal resolution and aspect ratio, and I would guess is that setting is rarely used by users.
That said, I expected people to disagree with the choice of only snapping to the XFB size in Auto AR mode, so maybe I could change the AR check to account for most AR modes.

Reply:
applied the requested change